### PR TITLE
Retries

### PR DESCRIPTION
**Pull request by katendejericho5**
#### Summary of Changes (What does this PR do?)

I have implemented the Retries feature in the restAPI.dart file and it prompts an HTTP get request three times before timeout .

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [x] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/main.dart [hunk 1]
```diff
@@ -6,17 +8,28 @@ import 'package:app/widgets/widgets.dart';
...
 
+    // Pass all uncaught "fatal" errors from the framework to Crashlytics
+    FlutterError.onError = (errorDetails) {
+      FirebaseCrashlytics.instance.recordFlutterFatalError(errorDetails);
+    };
+    // Pass all uncaught asynchronous errors that aren't handled by the Flutter framework to Crashlytics
+    PlatformDispatcher.instance.onError = (error, stack) {
+      FirebaseCrashlytics.instance.recordError(error, stack, fatal: true);
+      return true;
+    };
```
Comment (NoahNsimbe) [inline mobile/lib/main.dart:31]: @katendejericho5 is this part of the retires???

Comment (katendejericho5) [reply mobile/lib/main.dart:31]: No, I realised later that I had created a new branch from the crashlytics branch which was a mistake , but I don't know if there's a way of fixing it

File: mobile/lib/main.dart [hunk 2]
```diff
@@ -6,17 +8,28 @@ import 'package:app/widgets/widgets.dart';
...
     );
 
+    // Pass all uncaught "fatal" errors from the framework to Crashlytics
+    FlutterError.onError = (errorDetails) {
+      FirebaseCrashlytics.instance.recordFlutterFatalError(errorDetails);
+    };
+    // Pass all uncaught asynchronous errors that aren't handled by the Flutter framework to Crashlytics
+    PlatformDispatcher.instance.onError = (error, stack) {
+      FirebaseCrashlytics.instance.recordError(error, stack, fatal: true);
+      return true;
```
Comment (NoahNsimbe) [inline mobile/lib/main.dart:30]: @katendejericho5 I think these changes are not part of this PR.

Comment (katendejericho5) [reply mobile/lib/main.dart]: i have resolved it 

> @katendejericho5 I think these changes are not part of this PR.

i have now resolved it

**Review by katendejericho5 — COMMENTED** [review]
ready for review

**Review by NoahNsimbe — APPROVED** [review]
LGTM
